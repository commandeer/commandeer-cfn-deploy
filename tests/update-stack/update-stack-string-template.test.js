const { readFileSync } = require('fs');
const mockCreateStack = require('../mocks/updateStack');
const lib = require('../..');

describe('cfn-deploy', () => {
  beforeAll(() => mockCreateStack.beforeAll());

  it('should successfully create new stack', (done) => {
    const events = lib({
      region: 'us-east-1',
      stackname: 'existing-stack',
      templateString: readFileSync('./tests/templates/simple-template.yaml').toString(),
    });
    events.on('ERROR', (err) => done(err));
    events.on('COMPLETE', (data) => {
      expect(data).toMatchSnapshot();
    });
    events.on('FINALLY', done);
  });

  afterAll(() => mockCreateStack.afterAll());
});
