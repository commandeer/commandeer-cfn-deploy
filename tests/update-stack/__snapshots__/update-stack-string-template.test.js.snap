// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cfn-deploy should successfully create new stack 1`] = `
Object {
  "Capabilities": Array [],
  "ChangeSetName": "cfn-deploy",
  "ChangeSetType": "UPDATE",
  "Description": "Created with cfn-deploy",
  "Parameters": Array [],
  "StackName": "existing-stack",
  "Tags": Array [],
  "TemplateBody": "Resources:
  S3Bucket:
    Type: AWS::S3::Bucket
",
}
`;

exports[`cfn-deploy should successfully create new stack 2`] = `
Object {
  "StackStatus": "UPDATE_COMPLETE",
}
`;
